draft-ietf-quic-transport.md

```diff
@@ -322,175 +322,356 @@ all field sizes are in bits.  When discussing individual bits of fields, the
+state to do so.
+
+A Public Reset packet contains:
+
+* Octet 0: 0x82 (Flags indicating long header and Public Reset packet type)
+* Octets 1-8: Echoed data (octets 1-8 of received packet)
+* Octets 9-12: Echoed data (octets 9-12 of received packet)
+* Octets 13-16: Version (server version)
+* Octets 17+: Public Reset Proof
+
+For a client that sends a connection ID on every packet, a Public Reset
+packet received in response can be simply interpreted as:
+
+* Octets 1-8: Connection ID (echoed)
+* Octets 9-12: Echoed packet number, which could be 1, 2, or 4 bytes depending
+  on the packet number size used by the client, followed by 0, 2, or 3
+  subsequent bytes from the client packet.
+* Octets 13-16: Version (server version)
+* Octets 17+: Public Reset Proof
+
+A Public Reset packet sent by a server indicates that it does not have the
+state necessary to continue with a connection.  In this case, the server will
+include the fields that prove that it originally participated in the connection
+(see {{public-reset-proof}} for details).
+
+Upon receipt of a Public Reset packet that contains a valid proof, a client
+MUST tear down state associated with the connection.  The client MUST then
+cease sending packets on the connection and SHOULD discard any subsequent
+packets that arrive. A Public Reset that does not contain a valid proof MUST be
+ignored.
+
+### Public Reset Proof
+
+Details to be added.
+
+
+## Cleartext Packets {#cleartext-packet}
+
+Cleartext packets are sent during the handshake prior to key negotiation. A
+Client Cleartext packet contains:
+
+* Octet 0: 0x83 (Flags indicating long header and client Cleartext packet type)
+* Octets 1-8: Connection ID (randomly chosen)
+* Octets 9-12: Packet number (low 4 octets, starts at a random 31-bit value)
+* Octets 13-16: Version
+* Octets 17+: Payload
+
+A Server Cleartext End-of-Handshake packet contains:
+
+* Octet 0: 0x84 (Flags indicating Long header and appropriate server Cleartext
+  packet type)
+* Octets 1-8: Connection ID (server-selected value)
+* Octets 9-12: Packet Number (low 4 octets, random 31-bit initial value)
+* Octets 13-16: Version (echoed)
+* Octets 17+: Payload
+
+Other Server Cleartext packets contain:
+
+* Octet 0: 0x85 (Flags indicating Long header and appropriate server Cleartext
+  packet type)
+* Octets 1-8: Connection ID (echoed)
+* Octets 9-12: Packet Number (low 4 octets, random 31-bit initial value)
+* Octets 13-16: Version (echoed)
+* Octets 17+: Payload
+
+The client MUST choose a random value and use it as the Connection ID until the
+server replies with a server-selected Connection ID. Server-selected Connection
+IDs are used after a successful handshake, cleanly distinguishing packets that
+use them from packets using client-selected Connection IDs. All packets
+including and following a Server Cleartext End-of-Handshake packet use a
+server-selected Connection ID, as described in {{connection-id}}.
```

Comment:
+1.  Define the fields, then describe how they're used by reference to those fields, not the underlying bytes.